e2e-polybft/e2e/bridge_test.go

```diff
@@ -959,18 +978,30 @@ func TestE2E_Bridge_ChildChainMintableTokensTransfer(t *testing.T) {
 			require.NoError(t, err)
 		}
 
-		blockNum, err := childEthEndpoint.BlockNumber()
-		require.NoError(t, err)
+		allSuccessful := false
+
+		for it := 0; it < numberOfAttempts && !allSuccessful; it++ {
+			blockNum, err := childEthEndpoint.BlockNumber()
+			require.NoError(t, err)
 
-		// wait a couple of sprints to finalize state sync events
-		require.NoError(t, cluster.WaitForBlock(blockNum+3*sprintSize, 2*time.Minute))
+			// wait a couple of sprints to finalize state sync events
+			require.NoError(t, cluster.WaitForBlock(blockNum+3*sprintSize, 2*time.Minute))
 
-		// check that balances on the child chain are correct
-		for i, receiver := range depositors {
-			balance := erc20BalanceOf(t, receiver, contracts.NativeERC20TokenContract, childchainTxRelayer)
-			t.Log("Balance before", balancesBefore[i], "Balance after", balance)
-			require.Equal(t, balance, balancesBefore[i].Add(balancesBefore[i], big.NewInt(amount)))
+			allSuccessful = true
+
+			// check that balances on the child chain are correct
+			for i, receiver := range depositors {
+				balance := erc20BalanceOf(t, receiver, contracts.NativeERC20TokenContract, childchainTxRelayer)
+				t.Log("Balance before", balancesBefore[i], "Balance after", balance)
+				if balance.Cmp(balancesBefore[i].Add(balancesBefore[i], big.NewInt(amount))) != 0 {
```

Comment:
Should we print outer iterator value as well?

```suggestion
				t.Log("Attempt", it+1, "Balance before", balancesBefore[i], "Balance after", balance)
```